routing/dht/routing.go

```diff
@@ -77,10 +79,23 @@ func (dht *IpfsDHT) PutValue(ctx context.Context, key u.Key, value []byte) error
 	return nil
 }
 
+// ValReceipt represents a dht value record that has been received from a given peer
+// it is used to track peers with expired records in order to correct them.
+type ValReceipt struct {
+	From peer.ID
+	Val  *pb.Record
+}
```

Comment:
but we didnt put a value to a peer. this is for values we are getting from other peers. 

> receipts in distributed protocols usually mean a message that the other side returns

thats exactly what it is.